### ISSUE TITLE
[K2] Methods implemented by delegation miss the `override` keyword and KDoc

### ISSUE THREAD

**Issue by atyrin** [Kotlin/dokka#3384]
```Kotlin
interface CookieJar {
    /**
     * Saves cookies
     */
    fun saveFromResponse(url: String)
}

class CookieJarImpl() : CookieJar {
    override fun saveFromResponse(url: String) {}
}

class JavaNetCookieJar private constructor(
    delegate: CookieJarImpl,
) : CookieJar by delegate
```

With K1 the `JavaNetCookieJar` got `override` keyword and kdoc from the interface
<img width="723" alt="image" src="https://github.com/Kotlin/dokka/assets/7689049/8d97de30-c3f7-41eb-8ff0-7c2653c43652">

---

In K2 both are missed:
<img width="674" alt="image" src="https://github.com/Kotlin/dokka/assets/7689049/27557bd6-724d-414e-bf70-813c02d7b471">

**Installation**
- Dokka version: 1.9.20

Parent: https://github.com/Kotlin/dokka/issues/3328

**Comment by IgnatBeresnev** [comment]
We consider K1's behaviour to be correct in this case, so there should be the `override` keyword.

**Comment by vmishenev** [comment]
It can be implemented on Dokka's side.
See K1 implementation for descriptors:
https://github.com/JetBrains/kotlin/blob/5fba2962b9b5f6cb105fcd81dbd5f259cdd0db1d/core/descriptors/src/org/jetbrains/kotlin/resolve/DescriptorUtils.java#L138
```kt
    public static boolean isOverride(@NotNull CallableMemberDescriptor descriptor) {
        return !descriptor.getOverriddenDescriptors().isEmpty();
    }
```

**Comment by vmishenev** [comment]
It would be nice to have https://youtrack.jetbrains.com/issue/KT-64833/Analysis-API-Members-implemented-by-delegation-have-no-overridden-symbols to implement it like in K1.

**Comment by IgnatBeresnev** [comment]
This has been unblocked and fixed upstream, just need to check it and write a test.

Note: in K2 if a class inherits a function from an interface, clicking on this function in the child should lead to the interface.

The PR #3440 can be resurrected, but not necessarily
